### PR TITLE
Removing use of ManageWarnings within the code and unit tests.

### PR DESCRIPTION
**Pull request by Katie-Howard**
Closes https://github.com/metoppv/mo-blue-team/issues/437

This PR removes the `ManageWarnings` functionality usage from IMPROVER codebase as this handled centrally by Pytest. Before we adopted Pytest, this functionality was very useful with unittest.

Testing:
 - [x] Ran tests and they passed OK

### CODE REVIEW

File: improver_tests/blending/calculate_weights_and_blend/test_WeightAndBlend.py
Comment (bayliffe) [inline improver_tests/blending/calculate_weights_and_blend/test_WeightAndBlend.py:386]: Added back using pytest.warns.

File: improver_tests/blending/weighted_blend/test_MergeCubesForWeightedBlending.py
Comment (bayliffe) [inline improver_tests/blending/weighted_blend/test_MergeCubesForWeightedBlending.py:86]: Added back using pytest.warns.

File: improver_tests/calibration/ensemble_calibration/test_ContinuousRankedProbabilityScoreMinimisers.py
Comment (bayliffe) [inline improver_tests/calibration/ensemble_calibration/test_ContinuousRankedProbabilityScoreMinimisers.py:620]: Added back using pytest.warns

File: improver_tests/calibration/ensemble_calibration/test_ContinuousRankedProbabilityScoreMinimisers.py
Comment (bayliffe) [inline improver_tests/calibration/ensemble_calibration/test_ContinuousRankedProbabilityScoreMinimisers.py:1077]: Added back using pytest.warns

File: improver_tests/calibration/reliability_calibration/test_ApplyReliabilityCalibration.py
Comment (bayliffe) [inline improver_tests/calibration/reliability_calibration/test_ApplyReliabilityCalibration.py:285]: Added back using pytest.warns.

File: improver_tests/calibration/reliability_calibration/test_ApplyReliabilityCalibration.py
Comment (bayliffe) [inline improver_tests/calibration/reliability_calibration/test_ApplyReliabilityCalibration.py:518]: Added back with pytest.warns

File: improver_tests/ensemble_copula_coupling/test_ConvertProbabilitiesToPercentiles.py
Comment (bayliffe) [inline improver_tests/ensemble_copula_coupling/test_ConvertProbabilitiesToPercentiles.py:150]: Added back using pytest.warns

File: improver_tests/ensemble_copula_coupling/test_ConvertProbabilitiesToPercentiles.py
Comment (bayliffe) [inline improver_tests/ensemble_copula_coupling/test_ConvertProbabilitiesToPercentiles.py:345]: Added back using pytest.warns

File: improver_tests/generate_ancillaries/test_GenerateTopographicZoneWeights.py
Comment (bayliffe) [inline improver_tests/generate_ancillaries/test_GenerateTopographicZoneWeights.py:446]: Added back using pytest.warns.

File: improver_tests/metadata/test_forecast_times.py
Comment (bayliffe) [inline improver_tests/metadata/test_forecast_times.py:193]: Added back using pytests.warn
Removed duplicate calculation of result.

File: improver_tests/nbhood/recursive_filter/test_RecursiveFilter.py
Comment (bayliffe) [inline improver_tests/nbhood/recursive_filter/test_RecursiveFilter.py:183]: Added back using pytest.warns

File: improver_tests/nowcasting/optical_flow/test_OpticalFlow.py
Comment (bayliffe) [inline improver_tests/nowcasting/optical_flow/test_OpticalFlow.py:615]: Added back in using following syntax:
```
        with warnings.catch_warnings():
            warnings.simplefilter(action="error", category=UserWarning)
```
Such that an error is raised if a UserWarning is raised and the test fails.

File: improver_tests/spotdata/test_NeighbourSelection.py
Comment (bayliffe) [inline improver_tests/spotdata/test_NeighbourSelection.py:623]: Added back using pytest.warns

File: improver_tests/utilities/cube_manipulation/test_compare_attributes.py
Comment (bayliffe) [inline improver_tests/utilities/cube_manipulation/test_compare_attributes.py:79]: Added back using pytest.warns

File: improver_tests/utilities/cube_manipulation/test_compare_coords.py
Comment (bayliffe) [inline improver_tests/utilities/cube_manipulation/test_compare_coords.py:77]: Added back using pytest.warns

File: improver_tests/utilities/cube_manipulation/test_sort_coord_in_cube.py
Comment (bayliffe) [inline improver_tests/utilities/cube_manipulation/test_sort_coord_in_cube.py:183]: Added back using pytest.warns

File: improver_tests/utilities/temporal/test_temporal.py
Comment (bayliffe) [inline improver_tests/utilities/temporal/test_temporal.py:337]: Added back using pytest.warns

File: improver_tests/utilities/test_InterpolateUsingDifference.py
Comment (bayliffe) [inline improver_tests/utilities/test_InterpolateUsingDifference.py:293]: Added back using pytest.warns

File: improver_tests/wind_calculations/wind_gust_diagnostic/test_WindGustDiagnostic.py
Comment (bayliffe) [inline improver_tests/wind_calculations/wind_gust_diagnostic/test_WindGustDiagnostic.py:176]: Added back using pytest.warns

**Review by bayliffe — DISMISSED** [review]
I've reviewed this in as much as I have checked the removals are as expected. I have reintroduced tests that were removed erroneously, and I have fixed formatting errors. This needs a thorough second review in light of the changes I have made.

File: improver_tests/nowcasting/optical_flow/test_OpticalFlow.py
Comment (MoseleyS) [inline improver_tests/nowcasting/optical_flow/test_OpticalFlow.py:923]: You've done this exactly as it was before, but it should be simplified to:
```suggestion
            self.plugin.process(self.cube1, self.cube2, boxsize=3)
```

File: improver_tests/regrid/test_AdjustLandSeaPoints.py
Comment (MoseleyS) [inline improver_tests/regrid/test_AdjustLandSeaPoints.py:50]: `IGNORED_MESSAGES` and `WARNING_TYPES` are no longer used and need to be deleted.

**Review by MoseleyS — CHANGES_REQUESTED** [review]
Looks very good. Spotted a couple of things to fix.

**Review by MoseleyS — APPROVED** [review]

**Pull request merged**
